modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/elasticsearch/snapshot/ClusterCleaner.java

```diff
@@ -0,0 +1,56 @@
+package com.enonic.xp.repo.impl.elasticsearch.snapshot;
+
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.BundleException;
+import org.osgi.framework.startlevel.BundleStartLevel;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.enonic.xp.event.Event;
+import com.enonic.xp.event.EventListener;
+import com.enonic.xp.repo.impl.RepositoryEvents;
+
+@Component(immediate = true)
+public class ClusterCleaner
+    implements EventListener
+{
+    private final static Logger LOG = LoggerFactory.getLogger( ClusterCleaner.class );
+
+    private final BundleContext bundleContext;
+
+    @Activate
+    public ClusterCleaner( final BundleContext bundleContext )
+    {
+        this.bundleContext = bundleContext;
+    }
+
+    @Override
+    public void onEvent( final Event event )
+    {
+        if ( event.isType( RepositoryEvents.RESTORE_INITIALIZED_EVENT_TYPE ) )
+        {
+            for ( Bundle bundle : bundleContext.getBundles() )
+            {
+                if ( bundle.adapt( BundleStartLevel.class ).getStartLevel() > 40 )
```

Comment:
40 is a magic number, that represents XP launcher start level. Find a way to un-hardcode it.